models/base.js

```diff
@@ -0,0 +1,67 @@
+const Promise = require("bluebird");
+const db = require("../server/db");
+
+class BaseModel {
+  constructor(info) {
+    
+    this._info = info
+
+    info.columns.map(name => {
+      this[name] = info.data[name]
+    })
+  }
+
+  static _checkModelValidity() {
+    this._checkForTableName()
+    this._checkForPrimaryId()
+  }
+
+  static _checkForTableName() {
+      if(!this.tableName) {
+        throw new Error("This model needs a tableName!")
+      }    
+  }
+
+  static _checkForPrimaryId() {
+      if(!this.primaryId) {
+        throw new Error("This model needs a primaryId!")
+      }    
+  }
+
+  static _getSingleResponse(objects, fulfill, reject) {
+    if (!objects || objects.length === 0) {
+      reject(new Error("nothing returned from db"))
+    } else {
+      // return class instance
+      let instance = new this(objects[0])
```

Comment:
Do we need to validate what the `info` object looks like? Is there a comment somewhere that specifies exactly what it is that we need in `info`? I am assuming info is everything associated with the "sub" class that we would be instantiating... I assume this is very Rails-esque... But is there complexity in knowing/documenting `info` arg?